src/plots/plots.js

```diff
@@ -522,6 +553,7 @@ plots.supplyDefaults = function(gd) {
     // relink functions and _ attributes to promote consistency between plots
     relinkPrivateKeys(newFullLayout, oldFullLayout);
 
+    // TODO may return a promise
     plots.doAutoMargin(gd);
```

Comment:
Ha. Good to know. Thanks for the clarification. Yes, leaving this for a separate PR is fine.